tensorboard/backend/context.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Request-scoped context."""
+
+from tensorboard.data import auth as auth_lib
+
+
+class _TensorBoardRequestContextKey:
+    pass
+
+
+# A `RequestContext` value is stored on WSGI environments under this key
+# (an arbitrary sentinel object).
+_WSGI_KEY = _TensorBoardRequestContextKey()
+
+
+class RequestContext:
+    """Container of request-scoped values.
+
+    This context is for cross-cutting concerns: authentication,
+    authorization, auditing, internationalization, logging, and so on.
+    It is not simply for passing commonly used parameters to functions.
+
+    `RequestContext` values are to be treated as immutable.
+
+    Fields:
+      auth: An `AuthContext`, which may be empty but is never `None`.
+    """
+
+    def __init__(self, *, auth=None):
+        """Create a request context.
+
+        Args:
+          See "Fields" on class docstring. All arguments are optional
+          and will be replaced with default values if appropriate.
+        """
+        self.auth = auth if auth is not None else auth_lib.AuthContext.empty()
```

Comment:
If we're making this immutable, possibly nicer to use `@property` so that somebody can't wantonly reassign this.